src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/PreferConstCharOverConstUnitStringAnalyzer.cs

```diff
@@ -0,0 +1,116 @@
+        private static readonly LocalizableString s_localizableDescription = new LocalizableResourceString(nameof(MicrosoftNetCoreAnalyzersResources.PreferConstCharOverConstUnitStringInStringBuilderDescription), MicrosoftNetCoreAnalyzersResources.ResourceManager, typeof(MicrosoftNetCoreAnalyzersResources));
+
+        internal static DiagnosticDescriptor Rule = DiagnosticDescriptorHelper.Create(RuleId,
+                                                                                      s_localizableTitle,
+                                                                                      s_localizableMessage,
+                                                                                      DiagnosticCategory.Performance,
+                                                                                      RuleLevel.IdeSuggestion,
+                                                                                      s_localizableDescription,
+                                                                                      isPortedFxCopRule: false,
+                                                                                      isDataflowRule: false);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+            context.RegisterOperationAction(AnalyzeSymbol, OperationKind.Invocation);
+        }
+
+        private static void AnalyzeSymbol(OperationAnalysisContext context)
+        {
+            IInvocationOperation invocationOperation = (IInvocationOperation)context.Operation;
+            if (invocationOperation.Arguments.Length < 1)
+            {
+                return;
+            }
+
+            // Check that the object is a StringBuilder
+            if (!context.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemTextStringBuilder, out INamedTypeSymbol? stringBuilderType))
+            {
+                return;
+            }
+
+            IMethodSymbol appendStringMethod = stringBuilderType
+                .GetMembers("Append")
+                .OfType<IMethodSymbol>()
+                .WhereAsArray(s =>
+                    s.Parameters.Length == 1 &&
+                    s.Parameters[0].Type.SpecialType == SpecialType.System_String)
+                .FirstOrDefault();
+            if (appendStringMethod is null)
+            {
+                return;
+            }
+
+            if (!invocationOperation.TargetMethod.Equals(appendStringMethod))
+            {
+                return;
+            }
+
+            ImmutableArray<IArgumentOperation> arguments = invocationOperation.Arguments;
+            IArgumentOperation firstArgument = arguments[0];
```

Comment:
Is there any situation in which Arguments might be empty even though the target method takes one parameter?

Reply:
I'm guarding against that higher up with:

``` csharp
                    if (invocationOperation.Arguments.Length < 1)
                    {
                        return;
                    }
```